packages/digit_dss/lib/utils/utils.dart

```diff
@@ -0,0 +1,187 @@
+    return false;
+  }
+}
+
+Widget buildMetric(
+  BuildContext context,
+  String label,
+  String value,
+  int index,
+  DashboardLocalization localizations,
+) {
+  return Column(
+    crossAxisAlignment: CrossAxisAlignment.center,
+    mainAxisAlignment: MainAxisAlignment.spaceAround,
+    mainAxisSize: MainAxisSize.max,
+    children: [
+      if (index > 2)
+        SizedBox(
+          width: MediaQuery.of(context).size.width / 3.6,
+          child: Divider(
+            indent: 0,
+            height: kPadding,
+            endIndent: 0,
+            thickness: 2,
+            color: DigitTheme.instance.mobileTheme.colorScheme.outline,
+          ),
+        ),
+      ConstrainedBox(
+        constraints:
+            BoxConstraints(maxWidth: MediaQuery.of(context).size.width / 5),
+        child: Text(
+          value,
+          style: DigitTheme.instance.mobileTheme.textTheme.bodyLarge,
+          maxLines: 2,
+          textAlign: TextAlign.center,
+        ),
+      ),
+      ConstrainedBox(
+        constraints:
+            BoxConstraints(maxWidth: MediaQuery.of(context).size.width / 3.6),
+        child: Align(
+          alignment: Alignment.center,
+          child: Text(
+            localizations.translate(
+              label,
+            ),
+            textAlign: TextAlign.center,
+            style: DigitTheme.instance.mobileTheme.textTheme.bodyMedium,
+          ),
+        ),
+      ),
+    ],
+  );
+}
+
+DashboardRequestModel getRequestModel({
+  required String visualizationCode,
+  required String visualizationType,
+  required int startDate,
+  required int endDate,
+}) {
+  return DashboardRequestModel(
+    aggregationRequestDto: AggregationRequestDto(
+        visualizationCode: visualizationCode,
+        visualizationType: visualizationType,
+        requestDate: RequestDate(
+          startDate: startDate,
+          endDate: endDate,
+          interval: DSSEnums.day.toValue(),
+          title: DSSEnums.home.toValue(),
+        )),
+    headers: DSSHeaders(
+      tenantId: DashboardSingleton().tenantId,
+    ),
+  );
+}
+
+Future<void> processDashboardConfig(
+  Map<String, List<String>> dashboardConfig,
+  int startDate,
+  int endDate,
+  Isar isar,
+  DateTime lastSelectedDate,
+  DashboardRemoteRepository dashboardRemoteRepo,
+  String actionPath,
+  String tenantId,
+  String projectId,
+) async {
+  for (var entry in dashboardConfig.entries) {
+    String visualizationType = entry.key;
+    List<String> visualizationCodes = entry.value;
+
+    for (String visualizationCode in visualizationCodes) {
+      await dashboardRemoteRepo.searchAndWriteToDB(
+        apiEndPoint: actionPath,
+        lastSelectedDate: lastSelectedDate,
+        query: DashboardRequestModel(
+            aggregationRequestDto: AggregationRequestDto(
+                visualizationType: visualizationType,
+                visualizationCode: visualizationCode,
+                filters: {},
```

Comment:
Why do we need to pass empty map? And same for empty string for moduleLevel and qyeryType

Reply:
These are required param for the API to work, and we do not have any values for moduleCode now,